### PR TITLE
feat: integration tests mariadb assertion

### PR DESCRIPTION
**Pull request by f0rmiga**
Asserts the mariadb service basic connectivity.

### CODE REVIEW

File: tests/integration/minibroker_test.go [hunk 1]
```diff
@@ -58,10 +71,65 @@ var _ = Describe("classes", func() {
...
+				pod, err := kubeClient.CoreV1().Pods(namespace).Create(ctx, obj.(*corev1.Pod), metav1.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				defer func() {
+					err := kubeClient.CoreV1().Pods(namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				}()
+
+				By("asserting the mariadb client completed successfully")
+				retry := 0
+				for {
```
Comment (mook-as) [inline tests/integration/minibroker_test.go:113]: Is there a reason we're not doing
```suggestion
				for retry := 0; ; retry++ {
```

File: tests/integration/minibroker_test.go [hunk 2]
```diff
@@ -58,10 +71,65 @@ var _ = Describe("classes", func() {
...
+			assert: func(instance *apiv1beta1.ServiceInstance, binding *apiv1beta1.ServiceBinding) {
+				By("rendering and loading the mariadb client template")
+				tmplPath := path.Join(testDir, "resources", "mariadb_client.tmpl.yaml")
+				values := map[string]interface{}{
+					"DatabaseVersion": "10.3",
+					"SecretName":      binding.Spec.SecretName,
+					"Command": []string{
+						"sh", "-c",
+						"mysql" +
+							" --host=${DATABASE_HOST}" +
```
Comment (mook-as) [inline tests/integration/minibroker_test.go:91]: I think if we called this [`MYSQL_HOST`](https://dev.mysql.com/doc/refman/8.0/en/environment-variables.html) instead we don't need it as a command line argument. Also `MYSQL_PWD`. Not sure there's anything for user and DB name, though, so I guess none of this matters.

File: tests/integration/minibroker_test.go (hunk 1 above)
Comment (f0rmiga) [reply tests/integration/minibroker_test.go:113]: Too much Bash on my head.

File: tests/integration/minibroker_test.go (hunk 2 above)
Comment (f0rmiga) [reply tests/integration/minibroker_test.go:91]: Yeah, we don't have a comprehensive mapping for all the arguments. I'd rather keep it this way for consistency and clarity.

File: tests/integration/minibroker_test.go (hunk 1 above)
Comment (f0rmiga) [reply tests/integration/minibroker_test.go]: Fixed with https://github.com/kubernetes-sigs/minibroker/pull/63/commits/e6bec0390fa0bd4ff65d4db38f75f5c0d9cb73f8.

**Pull request merged**